core/src/databases/table_schema.rs

```diff
@@ -72,34 +75,75 @@ impl TableSchema {
         Ok(Self(schema))
     }
 
-    pub fn to_sql_string(&self, table_name: &str) -> String {
-        let mut create_table = format!("CREATE TABLE {} (", table_name);
+    pub fn get_create_table_sql_string(&self, table_name: &str) -> String {
+        format!(
+            "CREATE TABLE \"{}\" ({})",
+            table_name,
+            self.0
+                .iter()
+                .map(|(name, field_type)| {
+                    let sql_type = match field_type {
+                        TableSchemaFieldType::Int => "INT",
+                        TableSchemaFieldType::Float => "REAL",
+                        TableSchemaFieldType::Text => "TEXT",
+                        TableSchemaFieldType::Bool => "BOOLEAN",
+                    };
+                    format!("\"{}\" {}", name, sql_type)
+                })
+                .collect::<Vec<_>>()
+                .join(", ")
+        )
+    }
 
-        for (name, field_type) in &self.0 {
-            let sql_type = match field_type {
-                TableSchemaFieldType::Int => "INT",
-                TableSchemaFieldType::Float => "REAL",
-                TableSchemaFieldType::Text => "TEXT",
-                TableSchemaFieldType::Bool => "BOOLEAN",
-            };
+    pub fn get_insert_sql(&self, table_name: &str) -> (String, Vec<&String>) {
+        let field_names: Vec<&String> = self.0.keys().collect();
+        (
+            format!(
+                "INSERT INTO \"{}\" ({}) VALUES ({});",
+                table_name,
+                field_names
+                    .iter()
+                    .map(|name| format!("\"{}\"", name))
+                    .collect::<Vec<String>>()
+                    .join(", "),
+                field_names
+                    .iter()
+                    .enumerate()
+                    .map(|(i, _)| format!("?{}", i + 1))
+                    .collect::<Vec<String>>()
+                    .join(", ")
+            ),
+            field_names,
+        )
+    }
 
-            create_table.push_str(&format!("{} {}, ", name, sql_type));
+    pub fn get_insert_params(
```

Comment:
Not a big fan that it returns a value here

Reply:
Why ? Values are great because they are generic but contain type info, they are likely well optimized and battle-tested, they are what we get from serde JSON and they implement ToSql. Why would we use something else ?

Reply:
yet does not represent the data really as it handles nested structures which is impossible here.

Reply:
having a custom type will let us handle the true/false <-> 1/0 conversion in ToSql / FromSql which is exactly where you would expect the translation

Reply:
> having a custom type will let us handle the true/false <-> 1/0 conversion in ToSql / FromSql which is exactly where you would expect the translation


that is fair 👍